app/models/property.rb

```diff
@@ -36,11 +36,19 @@
 ###
 class Property < ApplicationRecord
   belongs_to :term
+  after_update :update_term, if: -> { saved_change_to_label? || saved_change_to_source_uri? }
```

Comment:
This works as expected, the only problem is the property is being updated twice. Can't we use a `before_update` callback to assigned its `uri` to a new value before any changes to the property have been saved? Like this:

```ruby
before_update :update_term, if: -> { label_changed? || source_uri_changed? }

private

def update_term
  return if term.name == label && term.source_uri == source_uri

  term.update!(name: label, source_uri:)
  self.uri = term.uri
end
```

Reply:
@excelsior - updated (force pushed), thank you.